integration_tests/src/main/python/csv_test.py

```diff
@@ -563,3 +563,19 @@ def test_csv_prefer_date_with_infer_schema(spark_tmp_path):
 
     assert_gpu_and_cpu_are_equal_collect(lambda spark: spark.read.option("inferSchema", "true").csv(data_path))
     assert_gpu_and_cpu_are_equal_collect(lambda spark: spark.read.option("inferSchema", "true").option("preferDate", "false").csv(data_path))
+
+@allow_non_gpu('FileSourceScanExec')
```

Comment:
Can we split the fallback tests off from the non-fallback tests?  Allowing the FileSourceScanExec to not be on the GPU when we test the non-fallback case could potentially mask issues.